hazelcast/src/main/java/com/hazelcast/internal/management/dto/MapConfigDTO.java

```diff
@@ -16,100 +16,170 @@
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
 import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.util.StringUtil;
 
 import java.io.IOException;
 
 import static com.hazelcast.util.JsonUtil.getBoolean;
 import static com.hazelcast.util.JsonUtil.getInt;
+import static com.hazelcast.util.JsonUtil.getObject;
 import static com.hazelcast.util.JsonUtil.getString;
 
 /**
  * Serializable adapter for {@link com.hazelcast.config.MapConfig}
  */
 public class MapConfigDTO implements JsonSerializable, IdentifiedDataSerializable {
 
-    private MapConfig config;
+    private MapConfig mapConfig;
 
     public MapConfigDTO() {
     }
 
     public MapConfigDTO(MapConfig mapConfig) {
-        this.config = mapConfig;
+        this.mapConfig = mapConfig;
     }
 
+    @Override
     public JsonObject toJson() {
         JsonObject root = new JsonObject();
-        root.add("name", config.getName());
-        root.add("memoryFormat", config.getInMemoryFormat().toString());
-        root.add("backupCount", config.getBackupCount());
-        root.add("asyncBackupCount", config.getAsyncBackupCount());
-        root.add("ttl", config.getTimeToLiveSeconds());
-        root.add("maxIdle", config.getMaxIdleSeconds());
-        root.add("maxSize", config.getMaxSizeConfig().getSize());
-        root.add("maxSizePolicy", config.getMaxSizeConfig().getMaxSizePolicy().toString());
-        root.add("readBackupData", config.isReadBackupData());
-        root.add("evictionPolicy", config.getEvictionPolicy().name());
+
+        String name = mapConfig.getName();
+        if (!StringUtil.isNullOrEmpty(name)) {
+            root.add("name", name);
+        }
+
+        String quorumName = mapConfig.getQuorumName();
+        if (!StringUtil.isNullOrEmpty(quorumName)) {
+            root.add("quorumName", quorumName);
+        }
+
+        root.add("maxSize", mapConfig.getMaxSizeConfig().getSize());
+        root.add("maxSizePolicy", mapConfig.getMaxSizeConfig().getMaxSizePolicy().toString());
+        root.add("evictionPolicy", mapConfig.getEvictionPolicy().name());
+        root.add("memoryFormat", mapConfig.getInMemoryFormat().toString());
+        root.add("cacheDeserializedValues", mapConfig.getCacheDeserializedValues().toString());
+        root.add("metadataPolicy", mapConfig.getMetadataPolicy().toString());
+        root.add("backupCount", mapConfig.getBackupCount());
+        root.add("asyncBackupCount", mapConfig.getAsyncBackupCount());
+        root.add("ttl", mapConfig.getTimeToLiveSeconds());
+        root.add("maxIdle", mapConfig.getMaxIdleSeconds());
+        root.add("readBackupData", mapConfig.isReadBackupData());
+        root.add("statisticsEnabled", mapConfig.isStatisticsEnabled());
+        root.add("mergePolicy", new MergePolicyConfigDTO(mapConfig.getMergePolicyConfig()).toJson());
+        root.add("mapStoreConfig", new MapStoreConfigDTO(mapConfig.getMapStoreConfig()).toJson());
```

Comment:
How about these?
```
    private WanReplicationRef wanReplicationRef;
    private List<EntryListenerConfig> entryListenerConfigs;
    private List<MapPartitionLostListenerConfig> partitionLostListenerConfigs;
    private List<MapIndexConfig> mapIndexConfigs;
    private List<MapAttributeConfig> mapAttributeConfigs;
    private List<QueryCacheConfig> queryCacheConfigs;
    private PartitioningStrategyConfig partitioningStrategyConfig;
    private HotRestartConfig hotRestartConfig = new HotRestartConfig();
    private MerkleTreeConfig merkleTreeConfig = new MerkleTreeConfig();
    private EventJournalConfig eventJournalConfig = new EventJournalConfig();
```

Reply:
yes, i noticed those ones, but since so many configs are there, opted to add them gradually when really needed.